OutOfSchool/OutOfSchool.WebApi/Program.cs

```diff
@@ -1,24 +1,52 @@
 using System;
-using System.Collections.Generic;
-using System.Linq;
-using System.Threading.Tasks;
 using Microsoft.AspNetCore.Hosting;
 using Microsoft.Extensions.Configuration;
 using Microsoft.Extensions.Hosting;
-using Microsoft.Extensions.Logging;
-using OutOfSchool.WebApi;
+using Microsoft.IdentityModel.Protocols;
+using Serilog;
 
-namespace OutOfSchool
+namespace OutOfSchool.WebApi
 {
     public class Program
     {
         public static void Main(string[] args)
         {
-            CreateHostBuilder(args).Build().Run();
+            // Read Configuration from appSettings
+            var config = new ConfigurationBuilder()
+                .AddJsonFile("appsettings.Development.json")
+                .Build();
+            
+            // Initialize Logger
+            Log.Logger = new LoggerConfiguration()
+                .ReadFrom.Configuration(config, sectionName: "Logging")
+                .Enrich.FromLogContext()
+                .WriteTo.File(
+                    path: config.GetSection("Logging:FilePath").Value, 
+                    rollingInterval: RollingInterval.Day, 
+                    retainedFileCountLimit: 2, 
+                    fileSizeLimitBytes: null)
+                .WriteTo.Debug()
+                .WriteTo.Console()
+                .CreateLogger();
+            
+            try
+            {
+                Log.Information("Application has started.");
+                CreateHostBuilder(args).Build().Run();
+            }
+            catch (Exception ex)
+            {
+                Log.Fatal(ex, "Application failed to start.");
+            }
+            finally
+            {
+                Log.CloseAndFlush();
+            }
         }
-
+        
         public static IHostBuilder CreateHostBuilder(string[] args) =>
             Host.CreateDefaultBuilder(args)
+                .UseSerilog() // Uses Serilog instead of default .NET Logger
```

Comment:
this comment is quite obvious
we usually comment other cases, complex logic etc.